quickwit-index-config/src/default_index_config/default_config.rs

```diff
@@ -95,6 +97,30 @@ impl DefaultIndexConfigBuilder {
             }
         }
 
+        // Resolve tag fields
```

Comment:
There was a misunderstanding here. Apologies, the spec you went for absolutely makes sense but this is not really what we want.

Tags are based on field existing in the schema. Country etc. 
They can be of any type. (not necessarily string).

We want to add an extra field in the schema, with the reserved name "_tags". (we already have a reserved field called source).

The _tags uses a raw tokenizer and is indexed like the schema you gave below.

When we create the tantivy document, we want to populate this extra field using the tag fields as defined in the index config. We populate it with the "field:field_value" template you used  on extract_tags.

extract_tags on the other hand, becomes simpler. We only need to iterate through the terms of the _tags field.

Reply:
Do we still support multiple fields for tagging? I guess if so, the `_tags` ends up with a string value like
`field1:x, field1:y, field2:z` then when we get the term value, we split by `,` and add them to the split tags. we will have to deduplicate.